### PR TITLE
Private userdb

### PR DESCRIPTION
**Pull request by johanlundberg**
Depends on https://github.com/SUNET/eduid-userdb/pull/87

### CODE REVIEW

**Comment by johanlundberg** [comment]
Need to merge https://github.com/SUNET/eduid-common/pull/67 before this one passes tests

File: src/eduid_common/api/decorators.py [hunk 1]
```diff
@@ -29,45 +29,19 @@ def require_eppn_decorator(*args, **kwargs):
...
-        kwargs['user'] = user
-        return f(*args, **kwargs)
-    return require_dashboard_user_decorator
-
-
 def require_support_personnel(f):
     @wraps(f)
     def require_support_decorator(*args, **kwargs):
-        user = _get_user()
+        user = get_user(no_private_userdb=True)
```
Comment (fredrikt) [inline src/eduid_common/api/decorators.py]: Is this easier to parse as private_userdb=False?

File: src/eduid_common/api/utils.py [hunk 2]
```diff
@@ -92,24 +93,19 @@ def get_user(userdb, user_class=ProofingUser):
...
-    :type user: eduid_userdb.dashboard.user.DashboardUser
+    :type user: current_app.private_userdb.UserClass
     """
-    if isinstance(user, User) and not isinstance(user, DashboardUser):
-        # turn it into a DashboardUser before saving it in the dashboard private db
-        user = DashboardUser(data = user.to_dict())
-    current_app.dashboard_userdb.save(user)
+    if not isinstance(user, current_app.private_userdb.UserClass):
+        # turn it into a private userdb user class before saving it in the private db
+        user = current_app.private_userdb.UserClass(data=user.to_dict())
```
Comment (fredrikt) [inline src/eduid_common/api/utils.py]: I would prefer it if we had to be a little more explicit about casting users from User to SomeOtherUser to make it easier to not save the wrong user. In effect, I would prefer if this save function would raise an exception if it was not passed a current_app.private_userdb.UserClass instance.

File: src/eduid_common/api/decorators.py (hunk 1 above)
Comment (johanlundberg) [reply src/eduid_common/api/decorators.py]: The get_user code actually saves the user in the private database with the correct modified_timestamp and that is what we don't want to do if no_private_userdb=True.

I don't feel strongly for or against no_private_userdb=True vs private_userdb=False, so I'll change it.

File: src/eduid_common/api/utils.py (hunk 2 above)
Comment (johanlundberg) [reply src/eduid_common/api/utils.py]: Agreed.

**Pull request merged**